packages/devtools_app/lib/src/inspector/flutter/inspector_tree_flutter.dart

```diff
@@ -215,6 +219,18 @@ class _InspectorTreeState extends State<InspectorTree>
     _bindToController();
   }
 
+  @override
+  void didUpdateWidget(InspectorTree oldWidget) {
+    if (oldWidget.controller != widget.controller) {
+      final InspectorTreeControllerFlutter oldController = oldWidget.controller;
+      oldController?.client = null;
+      cancel();
```

Comment:
this fixes some really puzzling bugs... We weren't properly clearing old event listeners resulting in the inspector selection getting handled by the wrong tree and other fun issues.